### PR TITLE
Add a section about JavaScript

### PR DESCRIPTION
**Pull request by awelzel**

### CODE REVIEW

File: scripting/javascript.rst [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+from JavaScript to Zeek are passed by copy as new Zeek record values. Changes
+to the original JavaScript object will not be reflected within Zeek.
+In below example, the ``intel_item`` JavaScript object will be converted to
+a new :zeek:see:`Intel::Item` Zeek record which is then
+passed to the :zeek:see:`Intel::insert` function. Modifying properties of
+``intel_item`` after it has been inserted to the Intel data store has
+no impact.
+
+.. literalinclude:: js/intel-insert.js
+   :caption: hello.js
```
Comment (timwoj) [inline scripting/javascript.rst:217]:
```suggestion
   :caption: intel-insert.js
```

File: scripting/javascript.rst [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+JavaScript objects keep a pointer to the Zeek record they represent.
+Holding a JavaScript object referencing a Zeek record value
+will keep it alive within Zeek even if Zeek itself does not reference
+it anymore. Updates to fields in Zeek become visible within JavaScript.
+Updates to properties of such objects in JavaScript become visible in Zeek.
+
+On the other hand, normal JavaScript objects (`{}` or `Object()`) are passed
+from JavaScript to Zeek are passed by copy as new Zeek record values. Changes
+to the original JavaScript object will not be reflected within Zeek.
+In below example, the ``intel_item`` JavaScript object will be converted to
```
Comment (timwoj) [inline scripting/javascript.rst:210]:
```suggestion
In the example below, the ``intel_item`` JavaScript object will be converted to
```

**Comment by awelzel** [comment]
Read the docs branch:

https://docs.zeek.org/en/topic-awelzel-js-docs/scripting/javascript.html

**Review by timwoj — CHANGES_REQUESTED** [review]

File: scripting/javascript.rst [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+can be leveraged for data ingestion (with :ref:`raw reader <input-raw-reader>`
+reader providing flexibility to consume input from external programs as events).
+The :ref:`broker-framework` is popular for exchanging events between
+Zeek and an external program using WebSockets.
+The external program sometimes solely acts as a proxy between Zeek and another
+external system.
+
+JavaScript integration adds to the above by enabling Zeek to load JavaScript
+code directly, thereby allowing developers to use its rich ecosystem of
+builtin and third-party directly within Zeek.
```
[the remark this answers is not in the document]
Comment (awelzel) [reply scripting/javascript.rst:37]: > packages

That had said libraries at some point, so used that again. Thanks!

File: scripting/javascript.rst [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+
+If you previously wanted to start a `HTTP server`_ within Zeek, record Zeek
+event data on-the-fly to a `Redis`_ database, got scared at looking at
+:zeek:see:`ActiveHTTP`'s implementation (or annoyed that it eats all newlines
+in HTTP responses), you may want to give JavaScript a go!
+
+Builtin Plugin
+==============
+
+The external `ZeekJS`_ plugin is included with Zeek as an optional builtin plugin.
```
[the remark this answers is not in the document]
Comment (awelzel) [reply scripting/javascript.rst:47]: I think @sethhall originally termed them "builtin plugins". I've been using builtin - when mentioning bifs, I'll try to adapt to built-in.

**Review by timwoj — APPROVED** [review]

**Pull request merged**